### ISSUE TITLE
Hi Rocio，can the preprocessed data be shared？

### ISSUE THREAD

**Issue by heyang9** [Rocionightwater/ML-NIDS-for-SCADA#1]
I am working on a project on ICS security，but lack of experimental data. I hope to do the job based on your work. I noticed that there are only raw data here, so can the preprocessed data be shared? It will help a lot for me！

**Comment by heyang9** [comment]
Dear Rocio,

Thank you so much！ Sorry for not reading your results carefully. 

I spent some time reading Ian Turnipseed's master thesis and your keynote to understand the details of the dataset better.  But I couldn't find your master thesis online.  Is it readable now? If I publish a paper, I will let you know and cite you!

Thanks again.
Best wishes,
Yang

**Comment by heyang9** [comment]
Hi Rocio, 

I did some experiments on your results, and now hopes to verify its effectiveness. So I wonder what 

changes you made to increase the features from 17 to 42 or 46... In the raw dataset, we can know what 

an exact feature means, but now I couldn’t find out what it means.

Because I need to return those packets to the original condition,  I have to know the features' meaning.

Best,

Yang

**Comment by Rocionightwater** [comment]
Hi Yang,

The number of features increased because I used 4 different strategies to
deal with the missing values (when preprocessing the datasets): (1) Keeping
the last existent value, (2) adding zeros (in the place the missing data
is) and adding indicators (add a 1 per missing value), (3) Gaussian mixture
model and (4) Kmeans (to cluster the payloads depending on the missing
values they payloads have).
For (1) Keep last value strategy you will have the same amount of features
(except for those that are categorical values, and I transform them to one
hot encoding notation), for (2) Zeros and Indicators, you will have many
more features since you are adding the indicators, and for (3) GMM and (4)
Kmeans, the clusters are in one hot encoding notation.
In conclusion, after preprocessing you will always have more features than
when you didn't preprocess.

I hope this can help you,

Best regards,
Rocio

2018-01-27 15:49 GMT+01:00 Yang He <notifications@github.com>:
